src/agent/debuglet.ts

```diff
@@ -318,11 +389,35 @@ export class Debuglet extends EventEmitter {
         that.scheduleRegistration_(0 /* immediately */);
         that.emit('started');
       });
-
     });
   }
 
   /**
+   * isReady returns a promise that only resolved if the last breakpoint update
+   * happend within a duration (PROMISE_RESOLVE_CUT_OFF_IN_MILLISECONDS). This
+   * feature is mainly used in Google Cloud Function (GCF), as it is a
+   * serverless environment and we wanted to make sure debug agent always
+   * captures the snapshots.
+   */
+  isReady(): Promise<void> {
+    if (Date.now() < this.breakpointFetchedTimestamp +
+            PROMISE_RESOLVE_CUT_OFF_IN_MILLISECONDS) {
+      return Promise.resolve();
+    } else {
+      const result = this.breakpointFetched.get();
+      if (result) {
+        return result;
+      }
+      this.breakpointFetched.refresh();
+      (this.debuggeeRegistered.get() as Promise<void>).then(() => {
```

Comment:
If you want to state that `this.debuggeeRegistered.get()` is not `null`, prefer the syntax `this.debuggeeRegistered.get()!.then(...` instead of `(this.debuggeeRegistered.get() as Promise<void>).then(...`.